.github/workflows/publish.yml

```diff
@@ -1,107 +1,92 @@
 name: Publish to PyPI
 on:
-  push:
-    tags:
-      - 'v[1-9][0-9].[0-9]+.[0-9]+'
-    branches: [main]
+  release:
+    types: [published]
 
 jobs:
-  pypi_release:
-    runs-on: ubuntu-22.04
+  build:
+    runs-on: ubuntu-latest
     env:
-      POETRY_VERSION:  1.3.0
       PYTHON_VERSION: "3.10"
-    if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
+    outputs:
+      sdist-package-name: ${{ steps.build-package.outputs.sdist-package-name }}
+      bdist-package-name: ${{ steps.build-package.outputs.bdist-package-name }}
     steps:
-      #----------------------------------------------
-      #       check-out repo and set-up python     
-      #----------------------------------------------
-      - name: Check out repository
-        uses: actions/checkout@v4
-
-      - name: Set up Python ${{ env.PYTHON_VERSION }}
-        uses: actions/setup-python@v5
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
         with:
           python-version: ${{ env.PYTHON_VERSION }}
+      - id: build-package
```

Comment:
Is this build-package step equivalent of the poetry build? I'd be concerned if they end up being slightly different, or, This doesn't take into account the poetry lock file anymore.

Reply:
It does replace that step. It doesn't use the poetry lock file but it gets the required packages from `pyproject.toml` which is where poetry takes its instructions from to begin with.